src/reducer.js

```diff
@@ -149,9 +150,9 @@ export const timestampsReducer = (state = {}, { type, path }) => {
 const createDataReducer = (actionKey = 'data') => (state = {}, action) => {
   switch (action.type) {
     case SET:
-      return set(state, getDotStrPath(action.path), action[actionKey])
+      return set(getDotStrPath(action.path), action[actionKey], state)
```

Comment:
Not sure this is what you want either. This points to the string path as the original data (which will always be a string). This set will need to be part of a spread/assign with state as it exists currently.